src/wazuh_modules/vulnerability_detector/wm_vuln_detector.c

```diff
@@ -6783,71 +6716,42 @@ int wm_vuldet_insert_MSU(sqlite3 *db, vu_msu_entries *msu) {
 
 int wm_vuldet_request_hotfixes(sqlite3 *db, const char *agent_id) {
     int retval = OS_INVALID;
-    char request[OS_MAXSTR];
-    cJSON *obj = NULL;
-    cJSON *obj_it;
+    cJSON *hotfixes = NULL;
     sqlite3_stmt *stmt = NULL;
-    char *last_id;
-
-    if (last_id = wm_vuldet_get_hotfix_scan(agent_id), !last_id) {
-        retval = 1;
-        goto end;
-    }
-
-    snprintf(request, OS_SIZE_128, vu_queries[VU_HOTFIXES_REQUEST], agent_id, last_id);
 
-    if (wm_vuldet_wdb_request(request, OS_MAXSTR)) {
+    if (OS_INVALID == wm_vuldet_get_hotfixes(atoi(agent_id), &hotfixes)) {
         goto end;
     }
+    else {
```

Comment:
We could check, just like packages, the JSON pointer before using it 
`if (hotfixes)`

Reply:
Fixed